plugins/module_utils/foreman_helper.py

```diff
@@ -178,6 +181,28 @@ def __init__(self, **kwargs):
         super(TaxonomyMixin, self).__init__(foreman_spec=foreman_spec, **kwargs)
 
 
+class ParametersMixin(object):
+    def __init__(self, **kwargs):
+        self.entity_name = kwargs.pop('entity_name', self.entity_name_from_class)
+        foreman_spec = dict(
+            parameters=dict(type='list', elements='dict', options=parameter_ansible_spec, flat_name='{0}_parameters_attributes'.format(self.entity_name)),
+        )
+        foreman_spec.update(kwargs.pop('foreman_spec', {}))
+        super(ParametersMixin, self).__init__(foreman_spec=foreman_spec, **kwargs)
+
+    def run(self, **kwargs):
+        entity = self.lookup_entity('entity')
+        if not self.desired_absent:
+            if entity and 'parameters' in entity:
+                entity['parameters'] = parameters_list_to_str_list(entity['parameters'])
+            parameters = self.foreman_params.get('parameters')
+            if parameters is not None:
+                self.foreman_params['parameters'] = parameters_list_to_str_list(parameters)
+
+        self.set_entity('entity', entity)
```

Comment:
This looks superfluous. The lookup sets the entity, and from that on, we change it in place.